extensions/ql-vscode/src/remote-queries/analyses-results-manager.ts

```diff
@@ -72,21 +70,39 @@ export class AnalysesResultsManager {
 
   private async downloadSingleAnalysisResults(
     analysis: AnalysisSummary,
-    credentials: Credentials
+    credentials: Credentials,
+    publishResults: (analysesResults: AnalysisResults[]) => Promise<void>
   ): Promise<void> {
-    const artifactPath = await downloadArtifactFromLink(credentials, analysis.downloadLink);
+    const analysisResults: AnalysisResults = {
+      nwo: analysis.nwo,
+      status: 'InProgress',
+      results: []
+    };
+
+    this.analysesResults.push(analysisResults);
+    void publishResults(this.analysesResults);
 
-    let analysisResults: AnalysisResults;
+    let artifactPath;
+    try {
+      artifactPath = await downloadArtifactFromLink(credentials, analysis.downloadLink);
+    }
+    catch (e) {
+      void showAndLogErrorMessage(
+        `There was an issue when downloading the analysis resutls for ${analysis.nwo}: ${e.message}`,
+        { fullMessage: e.stack });
+      return;
```

Comment:
As an alternative to popping up a message box, logging and returning maybe it would be better to log and only pop up a single message box after all promises have completed that lists all analyses that could not be downloaded. I think this would be a better experience, especially if there are lots of analyses that failed.

I can see three ways of doing this:

1. this function can use the results pattern you prototyped in the other branch and the calling function can handle failures appropriately
2. this function can throw an error on failure. The calling function can use `Promise.allSettled` to handle to deal with successes and failures
3. Pass in an error collector callback to this function that keeps track of errors as they happen. After all requests are finished, the calling function can inspect the error collector to handle any errors that were found.

Any of these would be sufficient.

Reply:
[Promise.all](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) will reject on the first error and since we run the batches serially, we'd only see 1 notification.

However, you're right that this approach isn't ideal as it can possibly hide some errors, so I've restructured a bit - used `Promise.allSettled` and aggregated errors.